src/main/java/rx/subscriptions/CompositeSubscription.java

```diff
@@ -55,6 +55,9 @@ public synchronized boolean isUnsubscribed() {
      *          the {@link Subscription} to add
      */
     public void add(final Subscription s) {
+        if (s == null) {
+            throw new IllegalArgumentException("Added Subscription cannot be null.");
```

Comment:
It's better to use `NullPointerException`. Other null assertions in RxJava throws NPE.